### DISCUSSION TITLE
Is it possible to add SMS or WhatsApp as 2FA Provider?

### DISCUSSION THREAD

**Discussion by ariaseta** [logto-io/logto#5849] [Ideas]
Is it possible to add SMS or WhatsApp as 2FA Provider?

**Comment by xiaoyijun** [comment]
Hi @ariaseta , we're researching related features, SMS as 2FA is in our roadmap. 3rd-party social app verfication is still under research.

cc @Rany0101

**Reply by ariaseta** [reply]
any example code to integrate it by ourself? or where the code is?

**Reply by vitalybrandes** [reply]
As option you can add Vonage service as 3th party provider. They have WhatsApp .fb and sms out of the box
